sdks/go/pkg/beam/artifact/materialize.go

```diff
@@ -69,41 +74,100 @@ func newMaterialize(ctx context.Context, endpoint string, dependencies []*pipepb
 	return newMaterializeWithClient(ctx, jobpb.NewArtifactRetrievalServiceClient(cc), dependencies, dest)
 }
 
-func newMaterializeWithClient(ctx context.Context, client jobpb.ArtifactRetrievalServiceClient, dependencies []*pipepb.ArtifactInformation, dest string) ([]*jobpb.ArtifactMetadata, error) {
+func newMaterializeWithClient(ctx context.Context, client jobpb.ArtifactRetrievalServiceClient, dependencies []*pipepb.ArtifactInformation, dest string) ([]*pipepb.ArtifactInformation, error) {
 	resolution, err := client.ResolveArtifacts(ctx, &jobpb.ResolveArtifactsRequest{Artifacts: dependencies})
 	if err != nil {
 		return nil, err
 	}
 
-	var md []*jobpb.ArtifactMetadata
+	var artifacts []*pipepb.ArtifactInformation
 	var list []retrievable
 	for _, dep := range resolution.Replacements {
 		path, err := extractStagingToPath(dep)
 		if err != nil {
 			return nil, err
 		}
-		md = append(md, &jobpb.ArtifactMetadata{
-			Name: path,
+		var filePayload pipepb.ArtifactFilePayload
+		if dep.TypeUrn != URNFileArtifact {
```

Comment:
Actually, this might be clearer constructing filePayload unconditionally based on Path, and then iff TypeUrn is URNFileArtifact, additionally setting the Sha256 attribute, if any.

Reply:
Done.